File: services/kodik-api.ts
const BASE_URL = 'https://neko-kodik.deno.dev'

export const searchKodikByShikimoriId = async (shikimoriId: number, withMaterialData: boolean = false) => {
  try {
    console.log('Searching anime in Kodik by Shikimori ID:', shikimoriId);
    
    const url = `${BASE_URL}/api/anime/?shikimori_id=${shikimoriId}${withMaterialData ? '&with_material_data=true' : ''}`;
    
    const response = await fetch(url);
    
    if (!response.ok) {
      throw new Error(`HTTP error! status: ${response.status}`);
    }
    
    const data = await response.json();

    if (data.results && Array.isArray(data.results) && data.results.length > 0) {
      const resultsWithAbsoluteLinks = data.results.map((result: any) => ({
        ...result,
        link: `${BASE_URL}/api/player/?id=${result.id}`
      }));
      return resultsWithAbsoluteLinks;
    }

    console.log('No results found for this Shikimori ID');
    return [];
  } catch (error) {
    console.error('Request error Kodik API:', error);
    throw new Error(`Request error Kodik API: ${error instanceof Error ? error.message : String(error)}`);
  }
};

File: lib/services/kodik_api.dart
// Порт `services/kodik-api.ts`. Параметры запроса — как в RN-версии,
// сменился только хост прокси (у старого деплоя на Deno кончился срок).
import 'dart:convert';
import 'package:http/http.dart' as http;

import '../models/kodik.dart';

const String kodikProxyUrl = 'https://neko-kodik.nezdanchick.deno.net';

Future<List<KodikInfo>> searchKodikByShikimoriId(
  int shikimoriId, {
  bool withMaterialData = false,
}) async {
  try {
    // ignore: avoid_print
    print('Searching anime in Kodik by Shikimori ID: $shikimoriId');

    final url = Uri.parse(
      '$kodikProxyUrl/api/anime/?shikimori_id=$shikimoriId'
      '${withMaterialData ? '&with_material_data=true' : ''}',
    );

    final response = await http.get(url);

    if (response.statusCode < 200 || response.statusCode >= 300) {
      throw Exception('HTTP error! status: ${response.statusCode}');
    }

    final data = jsonDecode(response.body) as Map<String, dynamic>;
    final results = data['results'];

    if (results is List && results.isNotEmpty) {
      return results.whereType<Map<String, dynamic>>().map((result) {
        final withLink = Map<String, dynamic>.from(result);
        withLink['link'] = '$kodikProxyUrl/api/player/?id=${result['id']}';
        return KodikInfo.fromJson(withLink);
      }).toList();
    }

    // ignore: avoid_print
    print('No results found for this Shikimori ID');
    return [];
  } catch (error) {
    // ignore: avoid_print
    print('Request error Kodik API: $error');
    throw Exception('Request error Kodik API: $error');
  }
}
